### PR TITLE
Multi Step Bar component

### PR DESCRIPTION
**Pull request by francoislg**
Fronts the component here: https://github.com/coveo/vapor/pull/504

### CODE REVIEW

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    tooltip?: ITooltipProps;
+}
+
+export interface IMultiStepBarProps {
+    className?: IClassName;
+    small?: boolean;
+    separator?: boolean;
+    inProgressAnimation?: 'slide' | 'shine';
+    flareAnimation?: boolean;
+    steps: IStep[];
```
Comment (GermainBergeron) [inline src/components/multiStepBar/MultiStepBar.tsx]: required params should be first in the interface

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    text?: string;
+    tooltip?: ITooltipProps;
+}
+
+export interface IMultiStepBarProps {
+    className?: IClassName;
+    small?: boolean;
+    separator?: boolean;
+    inProgressAnimation?: 'slide' | 'shine';
+    flareAnimation?: boolean;
```
Comment (GermainBergeron) [inline src/components/multiStepBar/MultiStepBar.tsx]: To keep the same logic as inProgressAnimation we should probably use `animation?: '' | 'flare';`

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    state: IMultiStepState;
+    text?: string;
+    tooltip?: ITooltipProps;
+}
+
+export interface IMultiStepBarProps {
+    className?: IClassName;
+    small?: boolean;
+    separator?: boolean;
+    inProgressAnimation?: 'slide' | 'shine';
```
Comment (GermainBergeron) [inline src/components/multiStepBar/MultiStepBar.tsx]: Should you allow empty string as a value?

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    return (
+        <div className={containerClasses}>
+            <div className='multi-step-bar-backdrop-container'>
+                {steps.map((step, i) => {
+                    const classes = classNames('multi-step-bar', `multi-step-bar-${step.state}`);
+                    return <div key={`multi-step-bar-backdrop-${i}`} className={classes}>{stepsContent[i]}</div>;
+                })}
+            </div>
+            <div className='multi-step-bar-content-container'>
+                {steps.map((step, i) => {
```
Comment (GermainBergeron) [inline src/components/multiStepBar/MultiStepBar.tsx]: Don't forget to type the method parameters

File: src/components/multiStepBar/tests/MultiStepBar.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        {state: 'done'},
+        {state: 'warning'},
+        {state: 'error'},
+        {state: 'doing'},
+        {state: 'to-do'},
+    ];
+    const stepsWithText: IStep[] = defaultSteps.map((step, i) => ({...step, text: `Step ${i}`}));
+    const testProps: IMultiStepBarProps = {steps: defaultSteps};
+
+    it('should render without error with multiple scenarios of number of steps and current step', () => {
```
Comment (GermainBergeron) [inline src/components/multiStepBar/tests/MultiStepBar.spec.tsx]: You don't seem to use multiple scenarios here

File: src/components/multiStepBar/MultiStepBar.tsx (hunk 3 above)
Comment (francoislg) [reply src/components/multiStepBar/MultiStepBar.tsx]: It is optional, so you can just omit it 🤔

But I guess we better put it just it case

**Comment by francoislg** [comment]
Closed in favor of https://github.com/coveo/react-vapor/pull/619

I have processed your reviews in the other branch :)

**Pull request closed without merging**